dev-util/visual-studio-code/visual-studio-code-1.22.2.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eutils xdg-utils
+
+DESCRIPTION="Microsoft's free and open-source code editor."
+HOMEPAGE="https://code.visualstudio.com/"
+SRC_URI="https://github.com/Microsoft/vscode/archive/${PV}.tar.gz -> ${P}.tar.gz
+	https://www.hlmjr.com/vscode-assets.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="net-libs/nodejs
+	sys-apps/yarn"
+RDEPEND="${DEPEND}
+	dev-libs/nss
+	dev-vcs/git
+	gnome-base/gconf
+	media-libs/alsa-lib
+	x11-libs/gtk+:2
+	x11-libs/libnotify
+	x11-libs/libxkbfile
+	x11-libs/libXScrnSaver
+	x11-libs/libXtst"
+
+S="${WORKDIR}"/vscode-"${PV}"
+
+pre_build_checks() {
+	CHECKREQS_MEMORY="2G" # due to max_old_space_size node setting below
+}
+
+src_prepare() {
+	eapply_user
+}
+
+src_compile() {
+	pushd "${S}"
+
+	# MS writes to different output directory based on arch w/different naming scheme
+	if use amd64; then
+		OUTPUT_ARCH=x64
+	elif use x86; then
+		OUTPUT_ARCH=ia32
+	fi
+
+	ulimit -n 8192 || die # avoid git errors >=1.22.0
+	yarn install --arch=${OUTPUT_ARCH}
+
+	# adjust max_old_space_size because default may be too low for node to build
+	/usr/bin/node --max_old_space_size=2048 node_modules/gulp/bin/gulp.js vscode-linux-${OUTPUT_ARCH}
+}
+
+src_install() {
+	# MS writes to different output directory based on arch w/different naming scheme
+	if use amd64; then
+		OUTPUT_ARCH=x64
+	elif use x86; then
+		OUTPUT_ARCH=ia32
+	fi
+
+	dosym "${EPREFIX}/opt/${PN}/bin/code-oss" "/usr/bin/code-oss"
+	domenu "${FILESDIR}/vscode.desktop"
+	doicon "${WORKDIR}/vscode.png"
+
+	insinto "/opt/${PN}"
+	doins -r "${WORKDIR}"/VSCode-linux-"${OUTPUT_ARCH}"/locales
+	doins -r "${WORKDIR}"/VSCode-linux-"${OUTPUT_ARCH}"/resources
+	doins "${WORKDIR}"/VSCode-linux-"${OUTPUT_ARCH}"/*.bin
+	doins "${WORKDIR}"/VSCode-linux-"${OUTPUT_ARCH}"/*.dat
+	doins "${WORKDIR}"/VSCode-linux-"${OUTPUT_ARCH}"/*.pak
+	doins "${WORKDIR}"/VSCode-linux-"${OUTPUT_ARCH}"/*.so
+
+	exeinto "/opt/${PN}"
+	doexe "${WORKDIR}/VSCode-linux-${OUTPUT_ARCH}/code-oss"
+	exeinto "/opt/${PN}/bin"
+	doexe "${WORKDIR}/VSCode-linux-${OUTPUT_ARCH}/bin/code-oss"
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+
+	elog "If you see the following error when loading a large workspace in VSCode:"
+	elog "    \"Code - OSS is running out of file handles.\""
+	elog "Raise the value of fs.inotify.max_user_watches in /etc/sysctl.conf and run:"
+	elog "    sysctl -p"
+	elog "https://code.visualstudio.com/docs/setup/linux#_error-enospc"
+}
+
+pkg_postrm() {
+	xdg_desktop_database_update
+}
+
+# TODO: Fix QA warnings (strip)
```

Comment:
`QA_PRESTRIPPED` is you friend

Reply:
Actually, it should be either `QA_PREBUILT` if it's prebuilt stuff, or the build system should be fixed not to strip.

Reply:
wait, it downloads something during build? it should not.. first of all, it's prohibited, also it will fail with network-sandbox. If it happens you should fetch that it tries to fetch yourself so it's in SRC_URI.

Reply:
Yes, looks that way. Yarn is basically an npm replacement that pulls down all the node modules it needs to build, and at the end of the gulp.js build, it downloads zipfiles of libnode (electron?) and libffmpeg.

Sounds like I might have to have packages for all the node dependencies before this can be pulled in. Please confirm, and I'll get back to working on that path.